File: 1394. Find Lucky Integer in an Array/solution.py
class Solution(object):
  def findLucky(self, arr):
    """
    :type arr: List[int]
    :rtype: int
    """
    dictionary = {}
    for num in arr:
      if num not in dictionary:
        dictionary[num] = 1
      else:
        dictionary[num] += 1
    
    maxValue = -1
    for num in dictionary:
      if num == dictionary[num] and dictionary[num] > maxValue:
        maxValue = dictionary[num]
    return maxValue

File: 1394. Find Lucky Integer in an Array/solution.cpp
class Solution {
public:
  int findLucky(vector<int>& arr) {
    unordered_map<int, int> dictionary;
    for (int index = 0; index < arr.size(); index++){
      int num = arr[index];
      dictionary[num]++;
    }
    
    int maxValue = -1;
    for (auto num : dictionary){
      if (num.first == num.second && num.second > maxValue){
        maxValue = num.second;
      }
    }
    return maxValue;
  }
};